### PR TITLE
ml_threshold int value fix,  data test refactored

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Fix ml_threshold check in case of integer value
- Data test refactored with counters. Use  ``--depth 3`` to link number of credential with existed tests

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest

### CODE REVIEW

File: credsweeper/app.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def _get_config_dict(
     # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
     def _use_ml_validation(self) -> bool:
-        if isinstance(self.ml_threshold, float) and self.ml_threshold <= 0:
+        if (isinstance(self.ml_threshold, float) or isinstance(self.ml_threshold, int)) and 0 >= self.ml_threshold:
```
Comment (babenek) [inline credsweeper/app.py:153]:
```suggestion
        if isinstance(self.ml_threshold, (float, int)) and 0 >= self.ml_threshold:
```

File: tests/test_main.py [hunk 2]
```diff
@@ -752,13 +750,14 @@ def prepare(report: List[Dict[str, Any]]):
...
         with tempfile.TemporaryDirectory() as tmp_dir:
-            for cfg in config:
-                # important key in .cfg.json is "json_filename"
+            for cfg in DATA_TEST_CFG:
                 with open(TESTS_PATH / "data" / cfg["json_filename"], "r") as f:
                     expected_result = json.load(f)
+                # informative parameter, relative with other tests counters
+                cred_count = cfg["__cred_count"]
+                # remove the value due CredSweeper does not know it
+                cfg.pop("__cred_count")
```
Comment (xDizzix) [inline tests/test_main.py:760]:
```suggestion
                cred_count = cfg.pop("__cred_count")
```

**Review by xDizzix — APPROVED** [review]
+

**Review by Yullia — APPROVED** [review]

**Review by csh519 — APPROVED** [review]
Thank you for refactoring.

LGTM 👍

**Pull request merged**